hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -1285,7 +1293,7 @@ private void sendMembershipEventNotifications(MemberImpl member, Set<Member> mem
             final MembershipServiceEvent event = new MembershipServiceEvent(membershipEvent);
             for (final MembershipAwareService service : membershipAwareServices) {
                 // service events should not block each other
```

Comment:
I've thought the same thing while commenting.
I wonder what was the reasoning behind this comment? Will we break something by blocking?
If yes than we may consider a striped executor here